camayoc/tests/qpc/api/v1/utils.py

```diff
@@ -12,6 +13,108 @@
     FailedScanException,
     WaitTimeError,
 )
+from camayoc.qpc_models import (
+    Credential,
+    Scan,
+    Source
+)
+
+
+def get_source(source_type, cleanup):
+    """Retrieve a single network source if available from config file.
+
+    Retreive one network source from the config file.
+    Sources listed under the following section are assumed to be available
+    on demand and do not require their power state to be managed.
+
+    The expected configuration in the Camayoc's configuration file is as
+    follows::
+
+        qpc:
+        # other needed qpc config data
+            sources:
+                 - hosts:
+                       - '127.0.0.1'
+                   name: local
+                   type: network
+                   credentials:
+                       - root
+
+    The credential listed is assumed to be in the top level 'credentials'
+    section.
+
+    This source is meant to be used for tests where we do not care about the
+    results of the scan, for example tests that assert we can pause and restart
+    a scan.
+
+    :returns: camayoc.qpc_models.Source that has been created on server and has
+        all credentials listed in config file created and associtated with it.
+    """
+    cfg = config.get_config()
+    cred_list = cfg.get('credentials', [])
+    src_list = cfg.get('qpc', {}).get('sources', [])
+    config_src = {}
+    if not (src_list and cred_list):
+        return
+    for src in src_list:
+        if src.get('type') == source_type:
+            config_src = src
+    if config_src:
+        config_src.setdefault('credential_ids', [])
+        src_creds = config_src.get('credentials')
+        for cred in src_creds:
+            for config_cred in cred_list:
+                if cred == config_cred['name']:
+                    server_cred = Credential(
+                        cred_type=source_type,
+                        username=config_cred['username'],
+                    )
+                    if config_cred.get('password'):
+                        server_cred.password = config_cred['password']
+                    else:
+                        server_cred.ssh_keyfile = config_cred['sshkeyfile']
+
+                    server_cred.create()
```

Comment:
What if the same credential is listed twice in the config file? Will this raise an exception since it will try to create an already created source?

Reply:
Don't think so because they get unique names generated for them on the server. So it doesn't mater if they have the same username and password or sshkey, because the "name" on the server is unique.

Reply:
Imagine this scenario we already run some tests and the same credential name was created and not removed from the server. I've forgotten to do the cleanup for CLI the first time the tests were added for example.

That was what I had in mind but it need to let this fail and fix the cleanup or config file if we find this problem.